common/make/binary_executable_building.mk

```diff
@@ -0,0 +1,90 @@
+# Copyright 2020 Google Inc. All Rights Reserved.
```

Comment:
WDYT about the name of this file?

Other ideas I had were "c_building_common.mk", "c_cpp_building.mk", "native_code_building.mk", etc. But wasn't sure about referring to specific "C"/"C++" languages; as for "native_code" - it seems confusing to refer to the sandboxed NaCl/Wasm code as "native".

Reply:
Agree on not using "native". I think referring to c/cpp specifically is fine, but wasm does support more than that (NaCl doesn't). So I think the current name is the best (I think it could be shortened to `executable_building.mk` - are there executables that are not binary?).

Reply:
Done - renamed to executable_building.mk.